model/publishing/delivery/PublishingDeliveryService.php

```diff
@@ -75,19 +86,21 @@ public function publishDelivery(\core_kernel_classes_Resource $delivery)
 
     /**
      * @param \core_kernel_classes_Resource $delivery
-     * @return \common_report_Report|null
+     * @return \common_report_Report
+     * @throws \common_exception_Error
+     * @throws \common_exception_NotFound
      */
     public function syncDelivery(\core_kernel_classes_Resource $delivery)
     {
         $environments = $this->getEnvironments();
-
         /** @var QueueDispatcher $queueDispatcher */
         $queueDispatcher = $this->getServiceManager()->get(QueueDispatcher::SERVICE_ID);
-
+        $taskLog = $this->getTaskLogFromDelivery($delivery);
+        $queueDispatcher->setOwner($taskLog->getOwner());
```

Comment:
I think we don't have to set the owner here because this method is only called when `DeliveryUpdatedEvent` is triggered as far as I see, it is triggered only when the user applies some changes on the web interface, we will have the owner from the session.